### DISCUSSION TITLE
Insert line break

### DISCUSSION THREAD

**Discussion by lutzissler** [slab/quill#4244] [Q&A]
Hi,

I would like to insert a line break (`<br>`). I discovered `blots/break.js`, which seemed to be a good fit. I tried insert that with

```js
const delta = quill.insertEmbed(range.index, 'break', true, Quill.sources.USER)
```

But nothing happens and the returned delta is empty. What am I doing wrong?

**Comment by lutzissler** [comment]
Is there any chance to get an answer to this? I got a very good impression of Quill/Parchment from the website and docs and also from working on some test implementations, but without proper support for line breaks, it doesn’t seem to be usable. Actually, structuring the document model in rows and using \n as the separator seems like a design flaw to me, even though there must be reasoning behind that.

I wonder how deep one would have to dive into Parchment to make line breaks work. The docs and code (especially [the exmple with the `<br/>`](https://cdn.jsdelivr.net/npm/quill@2/dist/quill.js)] suggest they should be working, but a closer looks make me think any `<br>` is being converted into a linebreak and therefore into a ‘row break’, i.e. a `<p>` when translating back to HTML.

Or am I just missing something and linebreaks are already working?

**Reply by HaloElite** [reply]
I also don't know what the problem might be for them not implementing soft breaks. It's an obvios feature. Also the reason why I'm not upgrading to 2.0 cause I finally got it "working" for 1.3.7. So please implement the soft breaks!

**Reply by lkalinowski** [reply]
Funny thing is that in documentation we can find nice Guide called [Cloning Medium with Parchment](https://quilljs.com/docs/guides/cloning-medium-with-parchment). Then I checked in Medium and of course there is a soft break solution 😅  (https://help.medium.com/hc/en-us/articles/215194537-Using-the-story-editor#content-271 - info at the end in common questions)

**Comment by lutzissler** [comment]
One can use an empty span to create a visual newline effect: https://codepen.io/johndoe75/pen/vYwdPBY

I wonder if creating a blot that represents such an empty span would work. When doing `.getSemanticHTML()`, one could easily `.replace()` those empty spans by `<br>`, and same before initializing the editor.

I tried to get this work but no luck. The docs are so bad… there’s a lot of stuff written down but they don’t actually *explain* things. I will keep on guessing…

**Reply by vcallegarip** [reply]
I am facing the same issue.... getSemanticHTML() removes my <br /> inside my <div>. Any luck from you side?

**Comment by lutzissler** [comment]
For anyone looking into the same issue, I have decided to use another editor.

**Reply by Interject-VictorC** [reply]
Which one?

**Reply by lutzissler** [reply]
We decided to go with ProseMirror. It’s not a ‘plug and play’ editor but requires a lot of code to make it work, but especially that turned out to make it a perfect fit for our project which requires heavy customization anyway.
